### PR TITLE
[ci] Request dep-audit team review if TCB deps are changed

### PR DESCRIPTION
**Pull request by metajack**
This notifies the dep-audit team and requests their review when TCB deps are changed by a PR.

### CODE REVIEW

**Comment by metajack** [comment]
Actually, I might as well add in the review check at the same time. Will add that now.

**Comment by rexhoffman** [comment]
Ohh can we have the same for the contents of .circleci/ and .github/ to our team?

**Comment by metajack** [comment]
Yeah, that should be no problem once I finish a `require-review` action.

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by metajack** [comment]
@davidiw Can you aporove this to see if the dep notifier action flips to green once a team approval is in?  I'll remove this final commit before landing.

**Review by davidiw — APPROVED** [review]
This is not a real approval, just testing the flow

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A color_space 0.4.3 (direct third-party, crates.io)
    * features: [none]
```

**Comment by metajack** [comment]
Ok, this is ready for review now.

Due to GHA security limitations I cannot check team membership from the normal GITHUB_TOKEN, which means I could not implement this using team review requirements. To work around this, there is now a file `./.github/dep-auditors.json` which contains the list of usernames who are allowed to approve dependency changes. PRs which change TCB dependencies will request review from all the dependency auditors, and any approving review from a dependency auditor will allow the action to succeed.

**Comment by metajack** [comment]
@bmwill Is anything else needed for bors to enforce the action succeeds?

File: .github/dep-auditors.json
Comment (bmwill) [inline .github/dep-auditors.json:3]: Are you not a reviewer? :)

**Review by bmwill — APPROVED** [review]

**Comment by bmwill** [comment]
@metajack We'd need the action to run on the auto branch and then the action would need to tell which PR it came from to pull reviewers from to check that the right approvers have reviewed it.

**Comment by metajack** [comment]
I'll add enforcement in a future PR. Going to land this to make sure it's functioning properly first.

/land

File: .github/dep-auditors.json
Comment (mimoo) [reply .github/dep-auditors.json:3]: sounds like @bmwill is volunteering as well

**Pull request closed without merging**
